checks/remotesettings/latest_approvals.py

```diff
@@ -72,11 +72,20 @@ async def get_latest_approvals(
         previous = history[i + 1]
         after = previous["last_modified"]
         before = current["last_modified"]
+
+        # We are interested in the number of history entries between two approvals. The
+        # approval timestamps are part of the history object data (ie. `target` field)
+        # and are not indexed on the server. In order to reduce the cost of the request,
+        # we will prefilter the history entries by their own timestamp,
+        # assuming the history entries were created within 1sec after the target object
+        # modification (usually it's a few milliseconds).
```

Comment:
Can you think of a query to verify the distribution of time differences between objects and history entries? We could also just deploy the change and see how it goes.

Reply:
It would be something like that (*untested*)

```
SELECT age(last_modified, from_epoch(data->'target'->'data'->'last_modified'))
FROM objects
WHERE resource_name = 'history'
ORDER BY 1 DESC
LIMIT 100
``` 

Both timestamps are assigned in the same request/transaction when the change occurs, so 1000msec is already way too much. We could even specify 1min, it would greatly reduce the dataset to scan.

Reply:
I had to convince Postgres that the datatypes are correct:
```
SELECT age(last_modified, from_epoch((data->'target'->'data'->>'last_modified')::bigint))
FROM objects
WHERE resource_name = 'history'
ORDER BY 1 DESC
LIMIT 100;
```
However, the results are all more than two years, so apparently something isn't quite right. Anyway, this was just meant as a quick verification. I'm also fine with just testing this in production – the worst that can happen is a failing Poucave check, which seems like a risk we can take. :)

Reply:
> However, the results are all more than two years, so apparently something isn't quite right.

Oh, now I'm curious :) 

> [...]  the worst that can happen is a failing Poucave check, which seems like a risk we can take. :)

No, not even. The only consequence would be that the number of changes for a certain approval would be inaccurate :)